packages/controller/src/resources/ingress/externalDns.ts

```diff
@@ -47,8 +47,8 @@ export function ExternalDnsResources(
   let annotations = {};
   if (target == "gcp") {
     annotations = {
-      "iam.gke.io/gcp-service-account": state.config.config!.gcp!
-        .externalDNSServiceAccount
+      "iam.gke.io/gcp-service-account":
+        state.config.config?.gcp?.externalDNSServiceAccount
```

Comment:
again here, we should explicitly `assert()` before we use it to ensure we throw an error if it doesn't exist